library/src/main/java/com/schibsted/spain/barista/assertion/BaristaListAssertions.kt

```diff
@@ -28,16 +29,18 @@ object BaristaListAssertions {
         val recyclerMatcher = findRecyclerMatcher(listId)
         val listViewMatcher = findListViewMatcher(listId)
 
-        try {
-            Espresso.onView(recyclerMatcher).check(RecyclerViewItemCountAssertion(expectedItemCount))
-        } catch (noRecyclerMatching: NoMatchingViewException) {
+        withFailureHandler(spyFailureHandler) {
             try {
-                Espresso.onView(listViewMatcher).check(ListViewItemCountAssertion(expectedItemCount))
```

Comment:
Instead of wrapping the whole method with `withFailureHandler` you can instead do `onView(...).withFailureHandler(spyFailureHandler).check(...)` in the two Espresso invocations.

Both methods work, but this one is native to Espresso, whereas `withFailureHandler` is a hack we did for some cases where Espresso doesn't support a custom failure handler. Better to avoid hacks if we can ;)

Reply:
:+1: I will update my PR with your suggestion.